esp-hal/src/timer/systimer.rs

```diff
@@ -131,6 +132,9 @@ impl<'d> SystemTimer<'d> {
         #[cfg(soc_etm)]
         etm::enable_etm();
 
+        // Don't reset Systimer as it will break `current_time`, only enable it
+        PeripheralClockControl::enable(PeripheralEnable::Systimer);
```

Comment:
Looking at the C6's TRM, this should be done before the `etm::enable_etm();` call above.

Since this peripheral may still be in reset, the register writes in that method will be ignored.

Reply:
Good catch, thanks!